src/hooks/use-creator-storage.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo, useState } from 'react';
+import type { Quiz, Round, Question } from '../../types';
+import { isCSR } from '../utils/common';
+
+const STORAGE_KEY = 'pubQuizApp-admin-createQuiz';
+
+export type StoreQuestion = Partial<Question>;
+export type StoreRound = Partial<Omit<Round, 'questions'>> & {
+  questions?: StoreQuestion[];
+};
+export type StoreQuiz = Partial<Omit<Quiz, 'rounds'>> & {
+  rounds?: StoreRound[];
+};
+
+const normalizeQuiz = (data: Partial<StoreQuiz>) => {
+  let rounds = data.rounds ?? [];
+
+  if (Array.isArray(rounds)) {
```

Comment:
can we ensure from BE that rounds will be always an array

Reply:
Dont know, but I would assume that BE can always provide the value as an array. But it doesnt matter here because this code is used for quiz creation, so the server doesnt really come in to the play here, yet. I think at least